tests/mr/device/arena_mr_tests.cpp

```diff
@@ -533,6 +533,42 @@ TEST_F(ArenaTest, Defragment)  // NOLINT
   }());
 }
 
+TEST_F(ArenaTest, PerThreadAndStreamArenas)  // NOLINT
+{
+  auto const arena_size = superblock::minimum_size * 4;
+  arena_mr mr(rmm::mr::get_current_device_resource(), arena_size);
+  std::vector<std::thread> threads;
+  std::size_t num_threads{3};
+  auto view        = std::make_shared<rmm::cuda_stream_view>(rmm::cuda_stream_per_thread);
+  void* thread_ptr = mr.allocate(256, view->value());
+  threads.reserve(num_threads);
+  for (std::size_t i = 0; i < num_threads; ++i) {
+    threads.emplace_back(std::thread([&] {
+      cuda_stream stream{};
+      void* ptr = mr.allocate(32_KiB, stream);
+      mr.deallocate(ptr, 32_KiB, stream);
```

Comment:
And immediately release these allocations.

Reply:
yes we don't need the allocations, we just needed to force the superblock into an arena other then global so that the defrag would cause all the superblocks back into the global arena.